frontstage_api/controllers/secure_messaging_controllers.py

```diff
@@ -1,53 +1,144 @@
 import json
 import logging
 
-from flask import request
-from requests import Response
 from structlog import wrap_logger
 
 from frontstage_api import app
 from frontstage_api.common.request_handler import request_handler
-from frontstage_api.exceptions.exceptions import UnexpectedStatusCode
+from frontstage_api.decorators.api_error_handler import api_error_handler
+from frontstage_api.exceptions.exceptions import ApiError
 
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def get_messages_list(encoded_jwt):
-    logger.debug('Attempting to retrieve the messages list')
-    method = 'GET'
-    label = request.args.get('label')
+@api_error_handler
+def get_messages_list(encoded_jwt, label):
+    logger.debug('Attempting to retrieve the messages list', label=label)
     url = '{}&label={}'.format(app.config['MESSAGES_LIST_URL'], label)
     headers = {"Authorization": encoded_jwt}
-    response = request_handler(method, url, headers)
+    response = request_handler('GET', url, headers=headers, error_code='FA001')
 
-    # If the response status code is not 200 call exception to stop request
     if response.status_code != 200:
-        logger.error('Error retrieving the messages list', status_code=response.status_code)
-        raise UnexpectedStatusCode(method, url, response.status_code, response.content)
+        logger.error('Error retrieving the messages list', label=label, status_code=response.status_code)
+        raise ApiError('FA001')
 
-    logger.debug('Successfully retrieved the messages list')
-    return json.loads(response.text)
+    logger.debug('Successfully retrieved the messages list', label=label)
+    messages_json = {"messages": json.loads(response.text)}
+    return messages_json
 
 
+@api_error_handler
 def get_unread_message_total(encoded_jwt):
     logger.debug('Attempting to retrieve the unread message total')
-    method = 'GET'
     url = app.config['UNREAD_MESSAGES_TOTAL_URL']
     headers = {"Authorization": encoded_jwt}
-    response = request_handler(method, url, headers, fail=False)
+    response = request_handler('GET', url, headers=headers, error_code='FA002')
 
-    # If request handler has failed to generate a response return the error message
-    if not isinstance(response, Response):
-        return response
-
-    # If the response status code is not 200 return json error message
     if response.status_code != 200:
-        logger.error('Error retrieving the unread messages total', status_code=response.status_code)
-        return {
-            "error": "Unexpected status code returned",
-            "url": url
-        }
+        raise ApiError('FA002')
 
     logger.debug('Successfully retrieved the unread message total')
     return json.loads(response.text)
```

Comment:
Maybe return the extracted number here (.get('total')). So the returned response is {unread_messages_total: xx} instead of {unread_messages_total: {total: xx}}.

Reply:
good idea I'll add this